.changeset/light-games-clap.md

```diff
@@ -0,0 +1,6 @@
+---
+'@redocly/openapi-core': minor
+'@redocly/cli': minor
+---
+
+Fixed an issue where remove-unused-components would not remove schemas which had a top-level allOf and improved detection of unused schemas in no-unused-components rule.
```

Comment:
@tatomyr do we need this in a conventional commit type of format? (`fix: `)

Reply:
No, this is a description that goes to the changelog. We decided to follow the existing changelog format.